server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/GroupsRoutes.java

```diff
@@ -202,6 +208,59 @@ public HaltException removeMultipleGroup(Request request, Response response) thr
         return halt(HttpStatus.NO_CONTENT_204);
     }
 
+    public static class IsGroupExistStructure {
+        // Fields
+        public String address;
+        public String status;
+        public String reason;
+
+        // Constructor
+        public IsGroupExistStructure(String address, String status, String reason) {
+            this.address = address;
+            this.status = status;
+            this.reason = reason;
+        }
+
+        @Override
+
+        public String toString() {
+            return "Address: " + address + "\nStatus: " + status + "\nReason: " + reason;
+        }
+    }
+
+    public String isExist(Request request, Response response) throws RecipientRewriteTableException, JsonProcessingException {
+
+        List<MappingSource> currentGroupList = recipientRewriteTable.getSourcesForType(Mapping.Type.Group).collect(ImmutableList.toImmutableList());
+        Map<String, Boolean> mp = new HashMap<>();
+        for (int i = 0; i < currentGroupList.size(); i++) {
+            mp.put(currentGroupList.get(i).asMailAddressString(), true);
+        }
+        String jsonString = request.body();
+        List<String> groups = objectMapper.readValue(jsonString, new TypeReference<List<String>>() {});
+
+        //checking is this group exist or not.
+        IsGroupExistStructure[] result = new IsGroupExistStructure[groups.size()];
+        for (int i = 0; i < groups.size(); i++) {
+            String group = groups.get(i);
+            MailAddress groupAddress;
```

Comment:
Try to validate the email address. What if i give you an address with invalid domain or format. If you give me `not exists` reason then in the next step if i want to create the group then it will give me error.

If validation fails then set status as `Error` and give `reason` for it